platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js

```diff
@@ -26,62 +31,82 @@ export default class WebSocket {
     this.url = url
     this.readyState = WebSocket.CONNECTING
 
-    const socketTask = my.connectSocket({
+    my.connectSocket({
       url,
-      protocols: Array.isArray(protocols) ? protocols : [protocols],
-      tcpNoDelay: true
-    })
-
-    _socketTask.set(this, socketTask)
-
-    socketTask.onClose((res) => {
-      this.readyState = WebSocket.CLOSED
-      if (typeof this.onclose === 'function') {
-        this.onclose(res)
-      }
-    })
-
-    socketTask.onMessage((res) => {
-      if (typeof this.onmessage === 'function') {
-        this.onmessage(res)
+      fail: function fail(res) {
+        this._triggerEvent('error', res)
       }
     })
 
-    socketTask.onOpen(() => {
+    this._onopen = () => {
       this.readyState = WebSocket.OPEN
-      if (typeof this.onopen === 'function') {
-        this.onopen()
-      }
-    })
+      this._triggerEvent('open')
+    }
+    my.onSocketOpen(this._onopen)
 
-    socketTask.onError((res) => {
-      if (typeof this.onerror === 'function') {
-        this.onerror(new Error(res.errMsg))
+    this._onmessage = (res) => {
```

Comment:
Why rename to `_onmessage`? It combines two words, so i think `_onMessage` is better.

Reply:
OK